tests/push.t

```diff
@@ -591,9 +589,8 @@ https://github.com/glandium/git-cinnabar/issues/338
   remote: transaction abort!
   remote: rollback completed
   remote: abort: No such file or directory: '/nonexistent'
-  fatal: called `Result::unwrap()` on an `Err` value: Error { kind: UnexpectedEof, message: "failed to fill whole buffer" }
-  Run the command again with `git -c cinnabar.check=traceback <command>` to see the full traceback.
-  error: git-remote-hg died of signal 6
+  To hg::.*/push.t/repo (re)
+   ! [remote rejected] 846552c6f25c1b46e784f59d8249fb31afac2996 -> branches/default/tip (nothing changed on remote)
```

Comment:
Shouldn't the git log below change too?

Reply:
Oh, I think I see what's going on: since the first one overwrites the remote refs, even when the second one doesn't do anything, it still keeps the wrong refs from the first push. How about adding a `git remote update` to reset the remote to what it's supposed to have?